### PR TITLE
side nav changes for alerting

### PR DESCRIPTION
**Pull request by riysaxen-amzn**
### Description
* This PR adds side navigation changes to notification plugin when feature is enabled
 
### Issues Resolved

When feature is enabled

Image 7-22-24 at 8 44 AM
Image 7-22-24 at 8 45 AM

When feature is disabled

Image 7-18-24 at 9 57 AM (1)
Image 7-18-24 at 9 57 AM

 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/alerting-dashboards-plugin/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/plugin.tsx [hunk 1]
```diff
@@ -64,6 +77,82 @@ export class AlertingPlugin implements Plugin<AlertingSetup, AlertingStart, Aler
...
+      core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS['security-analytics'], [
+        {
+          id: PLUGIN_NAME,
+          category: DEFAULT_APP_CATEGORIES.detect,
+          showInAllNavGroup: false
+        }
+      ])
+
+      // channels route
+      core.application.register({
```
Comment (SuZhou-Joe) [inline public/plugin.tsx:98]: Actually I am not very sure if we should register all the sub items of alerting plugin into left navigation. If it is aligned with requirement, then we are good.

**Review by SuZhou-Joe — APPROVED** [review]

File: public/app.js [hunk 2]
```diff
@@ -26,6 +26,10 @@ export function renderApp(coreStart, params) {
     dataSourceEnabled: getDataSourceEnabled()?.enabled,
   };
 
+  const navProps = {
+    defaultRoute: defaultRoute,
```
Comment (amsiglan) [inline public/app.js:30]: `: defaultRoute` redundant

File: public/app.js [hunk 3]
```diff
@@ -45,9 +49,10 @@ export function renderApp(coreStart, params) {
             isDarkMode,
             notifications: coreStart.notifications,
             chrome: coreStart.chrome,
+            defaultRoute: defaultRoute,
```
Comment (amsiglan) [inline public/app.js:52]: `: defaultRoute` redundant

File: public/plugin.tsx [hunk 4]
```diff
@@ -64,6 +77,82 @@ export class AlertingPlugin implements Plugin<AlertingSetup, AlertingStart, Aler
...
+          id: MONITORS_NAV_ID,
+          parentNavLinkId: PLUGIN_NAME,
+        },
+        {
+          id: DESTINATIONS_NAV_ID,
+          parentNavLinkId: PLUGIN_NAME,
+        },
+      ];
+      core.chrome.navGroup.addNavLinksToGroup(
+        DEFAULT_NAV_GROUPS['security-analytics'],
```
Comment (amsiglan) [inline public/plugin.tsx:146]: can we ask them to do `security_analytics` or `securityAnalytics` so we can just use it like `DEFAULT_NAV_GROUPS.security_analytics`?

**Review by amsiglan — APPROVED** [review]

**Review by AWSHurneyt — DISMISSED** [review]

File: public/plugin.tsx (hunk 1 above)
Comment (riysaxen-amzn) [reply public/plugin.tsx:98]: there are more changes coming as per new mocks

File: public/plugin.tsx (hunk 4 above)
Comment (riysaxen-amzn) [reply public/plugin.tsx:146]: have asked to make a change

File: public/plugin.tsx [hunk 5]
```diff
@@ -64,6 +77,82 @@ export class AlertingPlugin implements Plugin<AlertingSetup, AlertingStart, Aler
...
+        {
+          id: PLUGIN_NAME,
+          category: DEFAULT_APP_CATEGORIES.detect,
+          showInAllNavGroup: true
+        }
+      ])
+      core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS['security-analytics'], [
+        {
+          id: PLUGIN_NAME,
+          category: DEFAULT_APP_CATEGORIES.detect,
```
Comment (SuZhou-Joe) [inline public/plugin.tsx:92]: We should use Configure category here.

File: public/plugin.tsx [hunk 6]
```diff
@@ -64,6 +77,82 @@ export class AlertingPlugin implements Plugin<AlertingSetup, AlertingStart, Aler
...
+          showInAllNavGroup: false
+        }
+      ])
```
Comment (SuZhou-Joe) [inline public/plugin.tsx:95]:
```suggestion
```
Seems AD will only be visible in observability nav group

File: public/plugin.tsx [hunk 7]
```diff
@@ -64,6 +77,82 @@ export class AlertingPlugin implements Plugin<AlertingSetup, AlertingStart, Aler
       },
     });
 
+    if (core.chrome.navGroup.getNavGroupEnabled()) {
+      // register applications with category and use case information
+      core.chrome.navGroup.addNavLinksToGroup(DEFAULT_NAV_GROUPS.observability, [
+        {
+          id: PLUGIN_NAME,
+          category: DEFAULT_APP_CATEGORIES.detect,
+          showInAllNavGroup: true
```
Comment (SuZhou-Joe) [inline public/plugin.tsx:86]:
```suggestion
```
Seems alerting should not display in all use case.

Comment (riysaxen-amzn) [reply public/plugin.tsx]: addressed

File: public/plugin.tsx (hunk 6 above)
Comment (riysaxen-amzn) [reply public/plugin.tsx]: addressed

File: public/plugin.tsx [hunk 8]
```diff
@@ -64,6 +77,75 @@ export class AlertingPlugin implements Plugin<AlertingSetup, AlertingStart, Aler
...
+        },
+        {
+          id: DESTINATIONS_NAV_ID,
+          parentNavLinkId: PLUGIN_NAME,
+        },
+      ];
+      core.chrome.navGroup.addNavLinksToGroup(
+        DEFAULT_NAV_GROUPS['security-analytics'],
+        navLinks
+      );
```
Comment (SuZhou-Joe) [inline public/plugin.tsx:141]:
```suggestion
```
alerting is removed from security-analytics use case.

File: public/plugin.tsx [hunk 9]
```diff
@@ -64,6 +77,75 @@ export class AlertingPlugin implements Plugin<AlertingSetup, AlertingStart, Aler
...
+          showInAllNavGroup: false
+        }
+      ])
+
+      // channels route
+      core.application.register({
+        id: ALERTS_NAV_ID,
+        title: 'Alerts',
+        order: 9070,
+        category: DEFAULT_APP_CATEGORIES.configure,
```
Comment (SuZhou-Joe) [inline public/plugin.tsx:95]:
```suggestion
        category: DEFAULT_APP_CATEGORIES.detect,
```

File: public/plugin.tsx [hunk 10]
```diff
@@ -64,6 +77,75 @@ export class AlertingPlugin implements Plugin<AlertingSetup, AlertingStart, Aler
...
+        mount: async (params: AppMountParameters) => {
+          return mountWrapper(params, "/dashboard");
+        },
+      });
+
+      core.application.register({
+        id: MONITORS_NAV_ID,
+        title: 'Monitors',
+        order: 9070,
+        category: DEFAULT_APP_CATEGORIES.configure,
```
Comment (SuZhou-Joe) [inline public/plugin.tsx:106]:
```suggestion
        category: DEFAULT_APP_CATEGORIES.detect,
```

File: public/plugin.tsx [hunk 11]
```diff
@@ -64,6 +77,75 @@ export class AlertingPlugin implements Plugin<AlertingSetup, AlertingStart, Aler
...
+        mount: async (params: AppMountParameters) => {
+          return mountWrapper(params, "/monitors");
+        },
+      });
+
+      core.application.register({
+        id: DESTINATIONS_NAV_ID,
+        title: 'Destinations',
+        order: 9070,
+        category: DEFAULT_APP_CATEGORIES.configure,
```
Comment (SuZhou-Joe) [inline public/plugin.tsx:117]:
```suggestion
        category: DEFAULT_APP_CATEGORIES.detect,
```

**Review by SuZhou-Joe — APPROVED** [review]

**Review by AWSHurneyt — COMMENTED** [review]
@riysaxen-amzn The code changes LGTM. Could you check on the cypress failures?

**Comment by riysaxen-amzn** [comment]
> @riysaxen-amzn The code changes LGTM. Could you check on the cypress failures?

passes locally, i hav re-triggered the CI, most of the failures were timeout issue

```
 PASS  public/pages/CreateMonitor/containers/DefineMonitor/utils/mappings.test.js
 PASS  public/pages/CreateMonitor/components/AnomalyDetectors/EmptyFeaturesMessage/EmptyFeaturesMessage.test.js (15.358 s)
  ● Console

    console.log
      We only deal with feature related error type in this page: undefined

      at getErrorMsg (public/pages/CreateMonitor/components/AnomalyDetectors/EmptyFeaturesMessage/EmptyFeaturesMessage.js:28:15)

    console.log
      We only deal with feature related error type in this page: undefined

      at ActionUI (public/pages/CreateMonitor/components/AnomalyDetectors/EmptyFeaturesMessage/EmptyFeaturesMessage.js:57:15)

A worker process has failed to exit gracefully and has been force exited. This is likely caused by tests leaking due to improper teardown. Try running with --detectOpenHandles to find leaks. Active timers can also cause this, ensure that .unref() was called on them.

Test Suites: 3 skipped, 138 passed, 138 of 141 total
Tests:       20 skipped, 711 passed, 731 total
Snapshots:   197 passed, 197 total
Time:        238.629 s
Ran all test suites.

```

**Review by AWSHurneyt — APPROVED** [review]

**Review by AWSHurneyt — APPROVED** [review]

**Review by sbcd90 — APPROVED** [review]

**Pull request merged**